sunpy/coordinates/frames.py

```diff
@@ -0,0 +1,374 @@
+    definition of the longitude, which is defined using
+    an offset which is a time-dependent scalar value.
+
+    Parameters
+    ----------
+    representation: `~astropy.coordinates.BaseRepresentation` or None.
+        A representation object. If specified, other parameters must
+        be in keyword form.
+    lon: `Angle` object.
+        The longitude for this object (``lat`` must also be given and
+        ``representation`` must be None).
+    lat: `Angle` object.
+        The latitude for this object (``lon`` must also be given and
+        ``representation`` must be None).
+    rad: `astropy.units.Quantity` object, optional, must be keyword.
+        This quantity holds the radial distance. If not specified, it is, by
+        default, the solar radius. Optional, must be keyword.
+
+    Examples
+    --------
+    >>> sc = SkyCoord(1*u.deg, 2*u.deg, 3*u.km, frame="heliographiccarrington",
+    dateobs="2010/01/01T00:00:30")
+    >>> sc
+    <SkyCoord (HelioGraphicCarrington): dateobs=2010-01-01 00:00:30,
+    lon=1.0 deg, lat=2.0 deg, rad=3.0 km>
+    >>> sc = SkyCoord([1,2,3]*u.deg, [4,5,6]*u.deg, [5,6,7]*u.km,
+    dateobs="2010/01/01T00:00:45", frame="heliographiccarrington")
+    >>> sc
+    <SkyCoord (HelioGraphicCarrington): dateobs=2010-01-01 00:00:45,
+    (lon, lat, rad) in (deg, deg, km)
+        [(1.0, 4.0, 5.0), (2.0, 5.0, 6.0), (3.0, 6.0, 7.0)]>
+    """
+
+    default_representation = SphericalWrap180Representation
+
+    _frame_specific_representation_info = {
+        'spherical': [RepresentationMapping('lon', 'lon', 'recommended'),
+                      RepresentationMapping('lat', 'lat', 'recommended'),
+                      RepresentationMapping('distance', 'rad', 'recommended')],
+
+        'sphericalwrap180': [RepresentationMapping('lon', 'lon', 'recommended'),
+                             RepresentationMapping('lat', 'lat', 'recommended'),
+                             RepresentationMapping('distance', 'rad', 'recommended')]
+        }
+
+    dateobs = TimeFrameAttributeSunPy()
+
+
+class Heliocentric(BaseCoordinateFrame):
+    """
+    A coordinate or frame in the Heliocentric system.
+    This frame may either be specified in Cartesian
+    or cylindrical representation.
+    Cylindrical representation replaces (x, y) with
+    (rho, psi) where rho is the impact parameter and
+    psi is the position angle in degrees.
+
+    Parameters
+    ----------
+    representation: `~astropy.coordinates.BaseRepresentation` or None.
+        A representation object. If specified, other parameters must
+        be in keyword form and if x, y and z are specified, it must
+        be None.
+    x: `Quantity` object.
+        X-axis coordinate, optional, must be keyword.
+    y: `Quantity` object.
+        Y-axis coordinate, optional, must be keyword.
+    z: `Quantity` object. Shared by both representations.
+        Z-axis coordinate, optional, must be keyword.
+    D0: `Quantity` object.
```

Comment:
In some places, I saw a mention of ``dsun`` - should make sure this is consistent